gateway/extractors.go

```diff
@@ -0,0 +1,27 @@
+package gateway
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/cespare/xxhash"
+)
+
+type defaultSourceExtractor struct{}
+
+func (f defaultSourceExtractor) ExtractSource(r *http.Request) (string, error) {
+
+	authHeader := r.Header.Get("Authorization")
+	if authHeader == "" {
+		return "default", nil
+	}
+
+	return fmt.Sprintf("%d", xxhash.Sum64([]byte(authHeader))), nil
+}
+
+type defaultTCPSourceExtractor struct{}
+
+func (f defaultTCPSourceExtractor) ExtractSource(r *http.Request) (string, error) {
+
+	return r.RemoteAddr, nil
```

Comment:
I wonder if we should extract headers as well? Like X-Forwarded-For. 

Also I wonder if this is really a tcp limiter because we already did the tcp accept / TLS handshake at this point.

Maybe we could drop the tcp and rename that differently.